src/main/java/appeng/client/gui/implementations/GuiFluidTerminal.java

```diff
@@ -0,0 +1,362 @@
+					NetworkHandler.instance().sendToServer( new PacketInventoryAction( InventoryAction.FILL_ITEM, slot.slotNumber, 0 ) );
+				}
+				else
+				{
+					this.container.setTargetStack( meSlot.getAEStack() );
+					if( meSlot.getAEStack() != null )
+					{
+						AELog.info( "mouse1 GUI STACK SIZE %s", meSlot.getAEStack().getStackSize() );
+					}
+					NetworkHandler.instance().sendToServer( new PacketInventoryAction( InventoryAction.EMPTY_ITEM, slot.slotNumber, 0 ) );
+				}
+			}
+			return;
+		}
+		super.handleMouseClick( slot, slotIdx, mouseButton, clickType );
+	}
+
+	@Override
+	protected void keyTyped( final char character, final int key ) throws IOException
+	{
+		if( !this.checkHotbarKeys( key ) )
+		{
+			if( character == ' ' && this.searchField.getText().isEmpty() )
+			{
+				return;
+			}
+
+			if( this.searchField.textboxKeyTyped( character, key ) )
+			{
+				this.repo.setSearchString( this.searchField.getText() );
+				this.repo.updateView();
+				this.setScrollBar();
+			}
+			else
+			{
+				super.keyTyped( character, key );
+			}
+		}
+	}
+
+	@Override
+	protected void mouseClicked( final int xCoord, final int yCoord, final int btn ) throws IOException
+	{
+		this.searchField.mouseClicked( xCoord, yCoord, btn );
+
+		if( btn == 1 && this.searchField.isMouseIn( xCoord, yCoord ) )
+		{
+			this.searchField.setText( "" );
+			this.repo.setSearchString( "" );
+			this.repo.updateView();
+			this.setScrollBar();
+		}
+
+		super.mouseClicked( xCoord, yCoord, btn );
+	}
+
+	public void postUpdate( final List<IAEFluidStack> list )
+	{
+		for( final IAEFluidStack is : list )
+		{
+			this.repo.postUpdate( is );
+		}
+
+		this.repo.updateView();
+		this.setScrollBar();
+	}
+
+	private void setScrollBar()
+	{
+		this.getScrollBar().setTop( 18 ).setLeft( 175 ).setHeight( this.rows * 18 - 2 );
+		this.getScrollBar().setRange( 0, ( this.repo.size() + this.perRow - 1 ) / this.perRow - this.rows, Math.max( 1, this.rows / 6 ) );
+	}
+
+	@Override
+	public Enum getSortBy()
+	{
+		return this.configSrc.getSetting( Settings.SORT_BY );
+	}
+
+	@Override
+	public Enum getSortDir()
+	{
+		return this.configSrc.getSetting( Settings.SORT_DIRECTION );
+	}
+
+	@Override
+	public Enum getSortDisplay()
+	{
+		return this.configSrc.getSetting( Settings.VIEW_MODE );
+	}
+
+	@Override
+	public void updateSetting( IConfigManager manager, Enum settingName, Enum newValue )
+	{
+		if( this.sortByBox != null )
+		{
+			this.sortByBox.set( this.configSrc.getSetting( Settings.SORT_BY ) );
+		}
+
+		if( this.sortDirBox != null )
+		{
+			this.sortDirBox.set( this.configSrc.getSetting( Settings.SORT_DIRECTION ) );
+		}
+
+		this.repo.updateView();
+	}
+
+	protected List<SlotFluidME> getMeFluidSlots()
+	{
+		return this.meFluidSlots;
+	}
+
+	@Override
+	protected boolean isPowered()
+	{
+		return this.repo.hasPower();
+	}
+
+	protected String getBackground()
+	{
+		return "guis/terminal.png";

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method and declare a constant for this value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3400)